commit/src/mmcs.rs

```diff
@@ -76,7 +76,46 @@ pub trait Mmcs<T: Send + Sync + Clone>: Clone {
         commit: &Self::Commitment,
         dimensions: &[Dimensions],
         index: usize,
-        opened_values: &[Vec<T>],
-        proof: &Self::Proof,
+        batch_opening: &BatchOpening<T, Self>,
     ) -> Result<(), Self::Error>;
 }
+
+/// A Batched opening proof.
+///
+/// Allows for opening multiple matrices at multiple locations at once.
+#[derive(Serialize, Deserialize, Clone)]
+// Enable Serialize/Deserialize whenever T supports it.
+#[serde(bound(serialize = "T: Serialize"))]
+#[serde(bound(deserialize = "T: DeserializeOwned"))]
+pub struct BatchOpening<T: Send + Sync + Clone, InputMmcs: Mmcs<T>> {
+    pub opened_values: Vec<Vec<T>>,
+    pub opening_proof: <InputMmcs as Mmcs<T>>::Proof,
+}
+
+impl<T: Send + Sync + Clone, InputMmcs: Mmcs<T>> BatchOpening<T, InputMmcs> {
+    /// Creates a new batch opening proof.
+    pub fn new(opened_values: Vec<Vec<T>>, opening_proof: <InputMmcs as Mmcs<T>>::Proof) -> Self {
+        Self {
+            opened_values,
+            opening_proof,
+        }
+    }
+
+    /// Deconstructs the batch opening proof into its components.
+    pub fn deconstruct(self) -> (Vec<Vec<T>>, <InputMmcs as Mmcs<T>>::Proof) {
+        let BatchOpening {
+            opened_values,
+            opening_proof,
+        } = self;
+        (opened_values, opening_proof)
```

Comment:
Nit: Returning the fields directly is clearer:
```suggestion
        (self.opened_values, self.opening_proof)
```